android/gcm/app/src/main/java/gcm/play/android/samples/com/gcmquickstart/MainActivity.java

```diff
@@ -90,11 +91,12 @@ protected void onPause() {
      * the Google Play Store or enable it in the device's system settings.
      */
     private boolean checkPlayServices() {
-        int resultCode = GooglePlayServicesUtil.isGooglePlayServicesAvailable(this);
+        GoogleApiAvailability apiAvailability = GoogleApiAvailability.getInstance();
+        int resultCode = apiAvailability.isGooglePlayServicesAvailable(this);
         if (resultCode != ConnectionResult.SUCCESS) {
-            if (GooglePlayServicesUtil.isUserRecoverableError(resultCode)) {
-                GooglePlayServicesUtil.getErrorDialog(resultCode, this,
-                        PLAY_SERVICES_RESOLUTION_REQUEST).show();
+            if (apiAvailability.isUserResolvableError(resultCode)) {
+                apiAvailability.getErrorDialog(this, resultCode, PLAY_SERVICES_RESOLUTION_REQUEST)
+                        .show();
             } else {
                 Log.i(TAG, "This device is not supported.");
```

Comment:
Could you change this to `Log` as a higher priority (`Log.w` or `Log.e`) and to `Toast` instead of finishing the activity?  The problem with calling `finish()` is that the app may appear to open and crash which would result in confused developers.